keycloak.d.ts

```diff
@@ -0,0 +1,351 @@
+     *
+     * This method returns or promise or may optionally take a callback function.
+     *
+     * @param {Function} callback Optional callback, if not using promises.
+     */
+    obtainFromClientCredentials (callback?: (err: Error, grant: Grant) => void, scopeParam?: string): Promise<Grant>
+
+    /**
+     * Ensure that a grant is *fresh*, refreshing if required & possible.
+     *
+     * If the access_token is not expired, the grant is left untouched.
+     *
+     * If the access_token is expired, and a refresh_token is available,
+     * the grant is refreshed, in place (no new object is created),
+     * and returned.
+     *
+     * If the access_token is expired and no refresh_token is available,
+     * an error is provided.
+     *
+     * @param {Grant} grant The grant object to ensure freshness of
+     */
+    ensureFreshness (grant: Grant): Promise<Grant>
+
+    /**
+     * Perform live validation of an `access_token` against the Keycloak server.
+     *
+     * @param {Token|String} token The token to validate.
+     * @param {Function} callback Callback function if not using promises.
+     *
+     * @return {boolean} `false` if the token is invalid, or the same token if valid.
+     */
+    validateAccessToken<T extends Token|string>(token: T): Promise<false|T>
+
+    /**
+     * Returns a user info JSON Object
+     * @param {Token|String} token
+     */
+    userInfo<T extends Token|string, C extends StandardClaims>(token: T): Promise<C>
+
+    /**
+     * Create a `Grant` object from a string of JSON data.
+     *
+     * This method creates the `Grant` object, including
+     * the `access_token`, `refresh_token` and `id_token`
+     * if available, and validates each for expiration and
+     * against the known public-key of the server.
+     *
+     * @param {String|GrantProperties} rawData The raw JSON string received from the Keycloak server or from a client.
+     * @return {Promise} A promise reoslving a grant.
+     */
+    createGrant(data: string|GrantProperties): Promise<Grant>
+
+    /**
+     * Validate the grant and all tokens contained therein.
+     *
+     * This method examines a grant (in place) and rejects
+     * if any of the tokens are invalid. After this method
+     * resolves, the passed grant is guaranteed to have
+     * valid tokens.
+     *
+     * @param {Grant} grant The grant to validate.
+     *
+     * @return {Promise} That resolves to a validated grant or
+     * rejects with an error if any of the tokens are invalid.
+     */
+    validateGrant(grant: Grant): Promise<Grant>
+
+    /**
+     * Validate a token.
+     *
+     * This method accepts a token, and returns a promise
+     *
+     * If the token is valid the promise will be resolved with the token
+     * 
+     * If the token is undefined or fails validation an applicable error is returned
+     * 
+     * @return {Promise} That resolve a token
+     */
+    validateToken(token: Token): Promise<Token>
```

Comment:
Are the changes here ok? The previous comments (taken from the JS source) seemed outdated,

Reply:
Looks fine code-wise, but @abstractj would have to say if the comments are correct.